applications/crossbar/src/crossbar_view.erl

```diff
@@ -826,27 +854,36 @@ get_results(#{databases := [Db|RestDbs]=Dbs
 handle_query_result(#{last_key := LastKey
                      ,mapper := Mapper
                      ,context := Context
-                     }=LoadMap, Dbs, Results, Limit) ->
+                     ,page_size := PageSize
+                     }=LoadMap
+                   ,Dbs
+                   ,Results
+                   ,Limit
+                   ) ->
     ResultsLength = erlang:length(Results),
 
-    {NewLastKey, JObjs} = last_key(LastKey, Results, Limit, ResultsLength),
+    {NewLastKey, JObjs} = last_key(LastKey, Results, Limit, ResultsLength, PageSize),
 
     case apply_filter(Mapper, JObjs) of
         {'error', Reason} ->
             LoadMap#{context => cb_context:add_system_error('datastore_fault', kz_term:to_binary(Reason), Context)};
         FilteredJObjs when is_list(FilteredJObjs) ->
             FilteredLength = length(FilteredJObjs),
-            lager:debug("db_returned: ~b passed_filter: ~p next_start_key: ~p", [ResultsLength, FilteredLength, NewLastKey]),
-            handle_query_result(LoadMap, Dbs, FilteredJObjs, FilteredLength, NewLastKey);
-        FilteredJObj ->
```

Comment:
I think `cb_multi_factor` is still returning a single `JObj` (a change from this PR https://github.com/2600hz/kazoo/pull/5904).

Also for master version of this PR can you add this case clause back, it seems it gets removed somehow.

Reply:
Was going off all the specs and this type: https://github.com/2600hz/kazoo/blob/4.3/applications/crossbar/src/crossbar_view.erl#L83

Good to know about multi_factor, will look into it.

Reply:
I only see cb_multi_factor using `{'mapper', crossbar_view:map_value_fun()}`...am I missing a mapper that folds into a single JSON object instead?

Reply:
Looks like it calls `merge_summary` after the results are mapped/returned.

Reply:
You right, I don't why I was thinking it returning single `JObj` or why @lazedo added that case clause at the first place.